src/features/defi/providers/foxy/components/FoxyManager/Overview/FoxyOverview.tsx

```diff
@@ -60,44 +65,57 @@ export const FoxyOverview: React.FC<FoxyOverviewProps> = ({
     selectHighestBalanceAccountIdByStakingId(state, highestBalanceAccountIdFilter),
   )
 
-  const accountFilter = useMemo(
-    () => ({ accountId: accountId ?? highestBalanceAccountId ?? '' }),
-    [accountId, highestBalanceAccountId],
-  )
-  const bip44Params = useAppSelector(state => selectBIP44ParamsByAccountId(state, accountFilter))
-  const { data: foxyBalancesData, isLoading: isFoxyBalancesLoading } = useFoxyBalances({
-    accountNumber: bip44Params?.accountNumber ?? 0,
-  })
   const translate = useTranslate()
 
-  const opportunity = useMemo(
-    () => (foxyBalancesData?.opportunities || []).find(e => e.contractAssetId === assetId),
-    [foxyBalancesData?.opportunities, assetId],
+  const defaultAccountId = useAppSelector(state => selectFirstAccountIdByChainId(state, chainId))
+  const maybeAccountId = useMemo(
+    () => accountId ?? highestBalanceAccountId ?? defaultAccountId,
+    [accountId, defaultAccountId, highestBalanceAccountId],
   )
+  useEffect(() => {
+    if (!maybeAccountId) return
+    handleAccountIdChange(maybeAccountId)
+  }, [handleAccountIdChange, maybeAccountId])
```

Comment:
Looks like this is our goer for all opportunities to fix the flicker bits - to be confirmed by @shapeshift/operations, same implementation as THOR savers